data_diff/database.py

```diff
@@ -321,14 +481,36 @@ def _query(self, sql_code: str):
             raise ConnectError(msg % (sql_code, e))
 
         if res and isinstance(res[0], bigquery.table.Row):
-            res = [tuple(self._canonize_value(v) for v in row.values()) for row in res]
+            res = [tuple(self._normalize_returned_value(v) for v in row.values()) for row in res]
         return res
 
     def to_string(self, s: str):
         return f"cast({s} as string)"
 
+    def close(self):
+        self._client.close()
+
+    def select_table_schema(self, path: DbPath) -> str:
+        schema, table = self._normalize_path(path)
+
+        return (
+            f"SELECT column_name, data_type, 6 as datetime_precision, 6 as numeric_precision FROM {self.dataset}.INFORMATION_SCHEMA.COLUMNS "
+            f"WHERE table_name = '{table}' AND table_schema = '{schema}'"
+        )
+
+    def normalize_value_by_type(self, value, coltype: ColType) -> str:
+        if isinstance(coltype, PrecisionType):
+            return self.to_string(f'FORMAT_TIMESTAMP("%F %H:%M:%E6S", {value})')
```

Comment:
Don't you need to interpolate the precision value here, if the source is < 6? If it's not possible, at least raising an exception like for Redshift 👍🏻

Reply:
Hmm yeah probably. I had bigquery disabled because of the issues with snowflake, I forgot to test it 😞